### PR TITLE
Filter out appservices from mau count

### LINKED ISSUE #8403: Appservices users that sync will be counted towards MAU limits
Which means that you might unhelpfully find you at your limit if you enable a certain encrypted bridges project. I don't believe we want to count appservice users towards the block total, so hopefully this can be filtered out.

### PR DESCRIPTION
**Pull request by Half-Shot**
This is an attempt to fix #8403, however i'm totally failing to find the correct join magic to make this work.

### CODE REVIEW

File: synapse/storage/databases/main/monthly_active_users.py [hunk 1]
```diff
@@ -41,7 +41,14 @@ async def get_monthly_active_count(self) -> int:
         """
 
         def _count_users(txn):
-            sql = "SELECT COALESCE(count(*), 0) FROM monthly_active_users"
+            # Ensure we only fetch native users
```
Comment (anoadragon453) [inline synapse/storage/databases/main/monthly_active_users.py:44]:
```suggestion
            # Exclude app service users
```
I wasn't quite sure what a 'native user' was.

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
I'm a little worried that in the past we seem to have tracked application service users specifically in monthly user stats (e.g. https://github.com/matrix-org/synapse/pull/7030).

Also this looks like it might break some reporting graphs: https://github.com/matrix-org/synapse/blob/837293c314b47e988fe9532115476a6536cd6406/synapse/app/homeserver.py#L639-L648

I'd like a comment from @neilisfragile here before we go any further.

Alternatively, we may be able to fix this at a higher abstraction level.

**Comment by Half-Shot** [comment]
>I'm a little worried that in the past we seem to have tracked application service users specifically in monthly user stats (e.g. #7030).

This doesn't break that. The metric `synapse_admin_mau_current_mau_by_service` will continue to report this, but what this change will do is ensure that synapse won't block login/sync/actions due to appservices eating into MAU limits, which sounds desirable to me.

If folks still want to determine the total number of users, they should use `sum(synapse_admin_mau_current_mau_by_service)`

(As an aside, this metric has been broken for a while anyway -- appservice users aren't tracked at all until they /sync, which means this reporting system was broken for the vast majority of services anyway from day 1)

**Review by anoadragon453 — APPROVED** [review]
Thanks for the explanation. I'd still like to get a once-over from Neil, but it looks good from my perspective code-wise.

**Review by neilisfragile — APPROVED** [review]
LGTM.

The only reason for the app service split out was an ask from EMS, so if it is not useful it can happily be ripped out. No need to delay this PR for that though.

**Comment by anoadragon453** [comment]
Sounds good, thanks all!

**Pull request merged**